src/Rest/TestRoute.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/**
+ * Endpoint for testing Routes.
+ *
+ * Example call:
+ * /wp-json/eightshift-forms/v1/test-route
+ *
+ * @package EightshiftForms\Rest
+ */
+
+declare(strict_types=1);
+
+namespace EightshiftForms\Rest;
+
+use EightshiftForms\Hooks\Filters;
+use EightshiftForms\Captcha\BasicCaptcha;
+use EightshiftForms\Exception\UnverifiedRequestException;
+use EightshiftForms\Integrations\Authorization\AuthorizationInterface;
+
+/**
+ * Class TestRoute
+ */
+class TestRoute extends BaseRoute implements Filters
+{
+
+	/**
+	 * Route slug
+	 *
+	 * @var string
+	 */
+	public const ENDPOINT_SLUG = '/test-route';
+
+	/**
+	 * Test salt used for this route.
+	 *
+	 * @var string
+	 */
+	public const TEST_SALT = '1234-test-salt';
+
+	/**
+	 * Mock of required parameters.
+	 *
+	 * @var array
+	 */
+	public const REQUIRED_PARAMETER_1 = 'required-param';
+	public const REQUIRED_PARAMETER_2 = 'required-param-2';
+	public const IRRELEVANT_PARAM     = 'irrelevant-param';
+
+	/**
+	 * Implementation of the Authorization obj.
+	 *
+	 * @var AuthorizationInterface
+	 */
+	protected $hmac;
+
+	/**
+	 * Basic Captcha object.
+	 *
+	 * @var BasicCaptcha
+	 */
+	protected $basicCaptcha;
+
+	/**
+	 * Construct object
+	 *
+	 * @param AuthorizationInterface $hmac          Authorization object.
+	 * @param BasicCaptcha           $basicCaptcha BasicCaptcha object.
+	 */
+	public function __construct(
+		AuthorizationInterface $hmac,
+		BasicCaptcha $basicCaptcha
+	) {
+		$this->hmac          = $hmac;
+		$this->basicCaptcha = $basicCaptcha;
+	}
+
+	/**
+	 * Method that returns rest response
+	 *
+	 * @param  \WP_REST_Request $request Data got from endpoint url.
+	 *
+	 * @return WP_REST_Response|mixed If response generated an error, WP_Error, if response
+	 *                                is already an instance, WP_HTTP_Response, otherwise
+	 *                                returns a new WP_REST_Response instance.
+	 */
+	public function routeCallback(\WP_REST_Request $request)
+	{
+
+		try {
+			$params = $this->verifyRequest($request);
+		} catch (UnverifiedRequestException $e) {
+			return rest_ensure_response($e->getData());
+		}
+
+		$params = $this->unsetIrrelevantParams($params);
+
+		$mockResponse = [
+			'message' => 'all good',
+			'received-params' => $params,
+		];
+
+		return \rest_ensure_response([
+			'code' => 200,
+			'data' => $mockResponse,
+		]);
+	}
+
+	/**
+	 * Define authorization salt used for request to response handler.
+	 *
+	 * @return string
+	 */
+	protected function getAuthorizationSalt(): string
```

Comment:
If these methods are used only in this class they should be `private` 😉

Reply:
It's inherited from parent class.